lib/reply.js

```diff
@@ -138,9 +138,30 @@ Reply.prototype.send = function (payload) {
 
   if (this[kReplySerializer] !== null) {
     payload = this[kReplySerializer](payload)
-  } else if (hasContentType === false || contentType.indexOf('application/json') > -1) {
-    if (hasContentType === false || contentType.indexOf('charset') === -1) {
+
+  // The indexOf below also matches custom json mimetypes such as 'application/hal+json' or 'application/ld+json'
+  } else if (hasContentType === false || contentType.indexOf('json') > -1) {
+    if (hasContentType === false) {
       this[kReplyHeaders]['content-type'] = CONTENT_TYPE.JSON
+    } else {
+      // If hasContentType === true, we have a JSON mimetype
+      if (contentType.indexOf('charset') === -1) {
+        // If we have simply application/json instead of a custom json mimetype
+        if (contentType.indexOf('/json') > -1) {
```

Comment:
Sorry to delay this even more, but I wonder if the following is observably faster:
```suggestion
        if (contentType.slice(-5) === '/json') {
```

Reply:
No problem, let me fire up a benchmark for this.

Reply:
No, it won't work. I forgot about `content-type: appplication/json; charset=utf-8`. The `.indexOf` is required.

Reply:
Good point. Btw, I've measured the performance, and it is virtually identical, so even if it was an option, there is no benefit to it.

Reply:
Hi @jsumners and @kibertoad, what about `.slice(11, 16)`?

Reply:
> Hi @jsumners and @kibertoad, what about `.slice(11, 16)`?

No. The `; charset=utf-8` is an optional part of the value. The `.indexOf` guards against this.